### PR TITLE
Switch to using ActiveJob

### PR DESCRIPTION
**Pull request by jcoyne**

### CODE REVIEW

File: curation_concerns-models/app/jobs/characterize_job.rb [hunk 1]
```diff
@@ -1,11 +1,11 @@
 class CharacterizeJob < ActiveFedoraIdBasedJob
-  def queue_name
-    :characterize
-  end
+  queue_as :characterize
 
-  def run
+  def perform(id)
+    puts "PERFORMING"
```
Comment (mjgiarlo) [inline curation_concerns-models/app/jobs/characterize_job.rb]: Cruft?

File: curation_concerns-models/lib/curation_concerns/models/resque.rb [hunk 2]
```diff
@@ -1,36 +0,0 @@
...
-
-      def initialize(default_queue_name)
-        @default_queue_name = default_queue_name
-      end
-
-      def push(job)
-        push_tries = 0
-        queue = job.respond_to?(:queue_name) ? job.queue_name : default_queue_name
-        begin
-          ::Resque.enqueue_to queue, MarshaledJob, Base64.encode64(Marshal.dump(job))
```
Comment (mjgiarlo) [inline curation_concerns-models/lib/curation_concerns/models/resque.rb]: Cool, do we no longer need this b64 crap?

File: curation_concerns-models/lib/curation_concerns/models/resque.rb [hunk 3]
```diff
@@ -1,36 +0,0 @@
...
-        rescue Redis::CannotConnectError
-          ActiveFedora::Base.logger.error 'Redis is down!'
-        rescue Redis::TimeoutError => error
-          ActiveFedora::Base.logger.warn "Redis Timed out.  Trying again! #{job.inspect}"
```
Comment (mjgiarlo) [inline curation_concerns-models/lib/curation_concerns/models/resque.rb]: Do you have concerns about losing this code, @cam156 ?

File: spec/support/features.rb [hunk 4]
```diff
@@ -8,12 +8,12 @@
 
   config.before(:each, type: :feature) do
     Warden.test_mode!
-    @old_resque_inline_value = Resque.inline
-    Resque.inline = true
+    # @old_resque_inline_value = Resque.inline
```
Comment (mjgiarlo) [inline spec/support/features.rb]: Do we want to keep these comments?

File: app/jobs/visibility_copy_worker.rb [hunk 5]
```diff
@@ -1,16 +1,9 @@
...
-  def queue_name
-    :permissions
-  end
-
-  attr_accessor :pid
-
-  def initialize(pid)
-    self.pid = pid
-  end
+class VisibilityCopyWorker < ActiveFedoraIdBasedJob
```
Comment (mjgiarlo) [inline app/jobs/visibility_copy_worker.rb]: With this class moving into `app/jobs/` should it be renamed from a worker to a job for consistency?

File: curation_concerns-models/app/jobs/characterize_job.rb (hunk 1 above)
Comment (jcoyne) [inline curation_concerns-models/app/jobs/characterize_job.rb]: aye

File: curation_concerns-models/lib/curation_concerns/models/resque.rb (hunk 2 above)
Comment (jcoyne) [inline curation_concerns-models/lib/curation_concerns/models/resque.rb]: nope, ActiveJob serializes to json I believe.

Comment (mjgiarlo) [inline curation_concerns-models/lib/curation_concerns/models/resque.rb]: Huzzah. :fireworks:

**Pull request merged**